### ISSUE TITLE
pauseTest() does not work as expected

### ISSUE THREAD

**Issue by Turbo87** [emberjs/ember-mocha#75]
Currently the `pauseTest()` helper doesn't work properly with mocha due to its test timeouts. If you call `this.timeout(0);` before using `pauseTest()` it will work, but obviously it would be better if this could be done automatically. 

Is it possible for us to overwrite the original `pauseTest()` helper and add one that first calls `this.timeout(0)` and the original code afterwards?

**Comment by rwjblue** [comment]
Yeah, we should be able to do that.

**Comment by alexlafroscia** [comment]
I've been playing around with this some, and have had trouble overriding the existing `pauseTest` helper to set the timeout.  However, introducing a new helper works.

What I did to override the existing helper was adding this to the `test-helper.js` for a demo app:

```
import { before } from 'mocha';

before(function() {
  const originalPauseTestHelper = Ember.Test._helpers.pauseTest.method;

  Ember.Test.registerAsyncHelper('pauseTest', () => {
    this.timeout(0);

    return originalPauseTestHelper();
  });
});
```

My hope was that setting the `timeout` in here would override the default value of `2000` and cause the test now to fail.  However, this doesn't work; I'm guessing that's the case because the context (`this`) of the `before` hook needs to be set up before the `it` callback is called, but in this case we're setting it after the test has already started.

What did work was setting up a new helper, like this:

```
import { before } from 'mocha';

before(function() {
  const originalPauseTestHelper = Ember.Test._helpers.pauseTest.method;

  Ember.Test.registerAsyncHelper('pauseMochaTest', (app, context) => {
    context.timeout(0);

    return originalPauseTestHelper();
  });
});
```

In this case, the tests passes the context into the helper, like:

```
pauseMochaTest(this);
```

So now it can set the timeout correctly, because it's accessing the context of the `it` callback instead of the context of the `before`.

I've tried to see if there's some way to access the `it` context from the `before` but it doesn't seem possible. How do you feel about "solving" this with a new helper?

**Comment by alexlafroscia** [comment]
On the other hand, it seems like if you return a promise from the `it` callback, Mocha just waits for it to resolve. In this cause, `pauseTest` returns a promise that doesn't ever resolve, so as long as you do

```js
return pauseTest();
```

it seems like the test doesn't time out.

**Comment by Turbo87** [comment]
I'd be very surprised by that. The default behavior in Mocha is to timeout after 2 sec when either the `done` callback is used or a Promise is returned. Not sure if the ember-mocha-adapter mucks around with that though...

**Comment by alexlafroscia** [comment]
@Turbo87 what do you find surprising? I'm happy to throw together a demo repo.

**Comment by Turbo87** [comment]
I find it surprising that it doesn't time out when returning a Promise, because that is definitely not the default behavior of Mocha itself.

**Comment by alexlafroscia** [comment]
Hmm, you're right. I guess I was mistaken about what I was seeing earlier when returning `pauseTest()`.

**Comment by caseywatts** [comment]
this feels silly to have to do this, but I just added a snippet to atom for this and I thought I'd share :)

```
  'timeout then pauseTest()':
    'prefix': 'pau',
    'body': 'this.timeout(0); return pauseTest();'
```

**Comment by Turbo87** [comment]
This has been solved with the new testing API released in v0.14.0

**Comment by caseywatts** [comment]
I'm updating the docs for this here:
https://github.com/emberjs/ember.js/pull/16740
